From b2a6b4e4c74473de289b625cb1872bfc54698d45 Mon Sep 17 00:00:00 2001
From: Allan Stephens <allan.stephens@windriver.com>
Date: Wed, 26 Oct 2011 15:57:26 -0400
Subject: [PATCH 090/183] tipc: Fix bug in broadcast link duplicate message
 statistics
Git-commit: b2a6b4e4c74473de289b625cb1872bfc54698d45
Patch-mainline: Merged into tipc-devel
References: bnc#797455

Modifies broadcast link so that it increments the "received duplicate
message" count if an incoming message cannot be added to the deferred
message queue because it is already present in the queue. (The aligns
broadcast link behavior with that of TIPC's unicast links.)

Signed-off-by: Allan Stephens <allan.stephens@windriver.com>
Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>
Signed-off-by: Erik Hugne <erik.hugne@ericsson.com>
Signed-off-by: Jiri Kosina <jkosina@suse.cz>

---
 net/tipc/bcast.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/net/tipc/bcast.c b/net/tipc/bcast.c
index e7df313..035b350 100644
--- a/net/tipc/bcast.c
+++ b/net/tipc/bcast.c
@@ -603,6 +603,8 @@ receive:
 
 	if (deferred)
 		bcl->stats.deferred_recv++;
+	else
+		bcl->stats.duplicates++;
 
 unlock:
 	tipc_node_unlock(node);
-- 
1.7.8.3

